src/components/Pagination/Pagination.tsx

```diff
@@ -14,59 +14,69 @@ export interface PaginationProps {
   pageCount: number;
   /** The current page selected. */
   currentPage: number;
+  /** A method that returns a Link component given the target page */
+  getPageHref?: (pageNumber: number) => string;
   /** The method to callback when an item is selected. Passes the selected page to the consuming app as an argument. */
-  onPageChange?: (selected) => void;
+  onPageChange?: (selected: number) => void;
 }
 
 /** Pagination component that provides a list of page items */
-export default class Pagination extends React.Component<PaginationProps> {
-  constructor(props: PaginationProps) {
-    super(props);
+const Pagination: React.FC<PaginationProps> = (props: PaginationProps) => {
+  const {
+    blockName,
+    className,
+    modifiers = [],
+    pageCount,
+    currentPage,
+    getPageHref,
+    onPageChange,
+  } = props;
+  if (getPageHref && onPageChange) {
+    console.warn(
+      "Both onPageHref and onPageChange are passed. Will default to using onPageHref"
+    );
   }
 
-  previousPage = evt => {
-    const { currentPage } = this.props;
-    evt.preventDefault ? evt.preventDefault() : (evt.returnValue = false);
-    if (currentPage > 1) {
-      this.selectPage(evt, currentPage - 1);
-    }
-  };
+  const changeUrls =
+    typeof getPageHref !== "undefined" && typeof getPageHref === "function";
```

Comment:
We might want to beef up the URL validation a bit since right now, it is possible to break the pagination if you navigate to something like this:  
`http://localhost:9001/?path=/story/pagination--pagination-url-href&page=x`
or 
`http://localhost:9001/?path=/story/pagination--pagination-url-href&page=-15`


![image](https://user-images.githubusercontent.com/6432596/111666231-e55ae080-87e9-11eb-9d94-d3ff51a6ef03.png)

Reply:
Happy to validate getPage() to show that it should be done in the passed-in onClick.  That will prevent the page=-15 (or page=500) error - but won't prevent page="asdfaw;elkjfa;sdf" error.  But I think that gets into Storybook routing functionality, and pretty no consumer app uses Storybook as its main routing engine, so an example of this is unlikely to be helpful. 

What I could do instead is perhaps suggest in documentation that when users try to navigate to a page that doesn't exist, it returns to page 1, does that work better?